### PR TITLE
(FormField) fix: guard against Uncaught TypeError

### PR DESCRIPTION
**Pull request by buberdds**
In v2.29.0 FormField crashes with `Uncaught TypeError: Cannot read properties of null (reading 'getRootNode')
    at onFocus (FormField.js:500:1)` when a field uses `autoFocus` attr.  

#### What does this PR do?
Prevents FormField errors

#### What testing has been done on this PR?
manual 

#### How should this be manually tested?
Update `components/FormField/stories/Simple.js` with 

```
    <Box align="center" pad="large">
      <FormField label="Label" htmlFor="select">
        <TextArea autoFocus />
      </FormField>
    </Box>
```

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### Any background context you want to provide?
Relates to [6562](https://github.com/grommet/grommet/pull/6562)

#### What are the relevant issues?
-
#### Screenshots (if appropriate)
-
#### Do the grommet docs need to be updated?
no
#### Should this PR be mentioned in the release notes?
yes 
#### Is this change backwards compatible or is it a breaking change?
yes

### CODE REVIEW

File: src/js/components/FormField/FormField.js [hunk 1]
```diff
@@ -494,6 +494,9 @@ const FormField = forwardRef(
         {...outerProps}
         style={outerStyle}
         onFocus={(event) => {
+          if (!formFieldRef.current) {
```
Comment (jcfilben) [inline src/js/components/FormField/FormField.js:497]: I'm thinking that even if `formFieldRef.current` is undefined we should still run line 504 `if (onFocus) onFocus(event);`.

Maybe we could change this section to be:
``` javascript
onFocus={(event) => {
  const root = formFieldRef.current?.getRootNode();
  if (root) {
    setFocus(
      containsFocus(formFieldRef.current) && shouldKeepFocus(root),
    );
  }
  if (onFocus) onFocus(event);
}}
```

**Review by jcfilben — COMMENTED** [review]
Thanks for reporting and filing a fix for this bug! Added a comment below

File: src/js/components/FormField/FormField.js (hunk 1 above)
Comment (buberdds) [reply src/js/components/FormField/FormField.js:497]: ahh yeah looks safer, thanks

File: src/js/components/FormField/FormField.js [hunk 2]
```diff
@@ -501,10 +501,13 @@ const FormField = forwardRef(
         {...outerProps}
         style={outerStyle}
         onFocus={(event) => {
-          const root = formFieldRef.current.getRootNode();
-          setFocus(
-            containsFocus(formFieldRef.current) && shouldKeepFocus(root),
-          );
+          const root =
+            formFieldRef.current && formFieldRef.current.getRootNode();
```
Comment (MikeKingdom) [inline src/js/components/FormField/FormField.js:505]: Could simplify this to
```jsx
const root = formFieldRef.current?.getRootNode();
```

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**